OOP_L4/ConstructionInstallationManagement/Building.cs

```diff
@@ -0,0 +1,82 @@
+﻿
+namespace ConstructionInstallationManagement
+{
+    public class Building
+    {
+        private static int nextuid;
+
+        static Building()
+        {
+            nextuid = DateTime.Now.Millisecond;
+        }
+
+        private protected Building()
+        {
+            _uid = GetNewUID();
+        }
+
+        private protected Building(int height, int storey, int entrances, int rooms, string name = "")
```

Comment:
```suggestion
        private protected Building(int Нeight, int Storey, int Entrances, int Rooms, string Name = "")
```

Разделяйте способ написания полей, свойств, локальных переменных и входных переменных. Так жить будет проще. И код не будет превращаться в кашу.